### DISCUSSION TITLE
Op Stask: Missing depositErc20To Method?

### DISCUSSION THREAD

**Discussion by songkeys** [wevm/viem#1646] [Question]
The `depositERC20` seems missing in viem@2.0.0 but available in [op-viem](https://github.com/base-org/op-viem/blob/882f1066c1ffc48c2881f0037a65017d6e8e4c0e/src/actions/wallet/L1/writeDepositERC20.ts#L41-L53).

Looks like we have to call `l1StandardBridge` contract instead of `portal` contract which is missing in opstack chain's definitions?

**Comment by gndelia** [comment]
I implemented the depositing/withdrawing of ETH/ERC20 tokens on a fork of OP or a webapp.

Using the Portal contract did not work, at least with the old version of OP, using the L1StandardBridge, and the L2Bridge. So I had to use a custom implementation for depositing and initiating withdrawals for ERC20. In case anyone needs it, [here is the deposit](https://github.com/hemilabs/ui-monorepo/blob/05e51c5965e70f72993332945c357ba1746ec5f2/packages/hemi-tunnel-actions/src/depositErc20.ts) and the [withdrawal](https://github.com/hemilabs/ui-monorepo/blob/main/packages/hemi-tunnel-actions/src/initiateWithdraw.ts).

We may eventually move this into a publishable package
